### DISCUSSION TITLE
Is it possible to run linting like pyright via command line to generate output for a project

### DISCUSSION THREAD

**Discussion by mani2106** [microsoft/pylance-release#883] [Q&A]
Pyright is available as an npm package and having both installed is not recommended as said in the documentation, so is there a way run the linting for a project as a whole?

**Comment by erictraut** [comment]
You said "having both installed is not recommended". What are you referring to when you say "both"? The Pyright CLI tool can be installed along with the Pyright VS Code extension. The Pyright CLI can also be installed with the Pylance VS Code extension. However, the two VS Code extensions (Pylance and Pyright) shouldn't both be installed.

If you want to run type analysis on your entire workspace interactively, the Pyright or Pylance VS Code extension can do that too. Just set the "python.analysis.diagnosticMode" setting to "workspace". By default, both extensions default to "openFilesOnly".

**Reply by jeska** [reply]
This may refer to the following line in the README for pyright (from [here](https://github.com/microsoft/pyright#using-pyright-with-vs-code-python-extension)):

`Installing both Pyright and Pylance at the same time is not recommended. If both are installed and enabled, you will see duplicate errors, hover text, and completion suggestions.`

Even though this is under the section for the extensions, I think it can be misinterpreted as saying that the pyright CLI tool and the Pylance VS Code extension should not co-exist. (I say this as I had to re-read it a couple times to make sure I wasn't misunderstanding!) Maybe the docs can be amended for clarity?

**Reply by erictraut** [reply]
That's a good point. The sentence you quoted above is actually somewhat obsolete because the Pyright VS Code extension now looks for the presence of Pylance and disables itself (reporting an error to the user), so it actively prevents running alongside Pylance. That means there's no longer a reason to include this sentence in the documentation. I've deleted it to eliminate further confusion.

**Reply by jeska** [reply]
Sounds good! Thanks for the quick response!
